src/Native/Unix/System.Native/pal_process.c

```diff
@@ -289,6 +306,11 @@ done:;
         CloseIfOpen(waitForChildToExecPipe[READ_END_OF_PIPE]);
     }
 
+#ifndef HAVE_PIPE2
+    if (have_create_process_lock)
+        pthread_mutex_unlock(&process_create_lock);
```

Comment:
Nit: The rest of this file has `{` / `}` ever around single single line statements.